Sources/MultiLevelListSelectionFilter/MultiLevelListSelectionFilterViewController.swift

```diff
@@ -232,24 +218,29 @@ extension MultiLevelListSelectionFilterViewController: UITableViewDelegate {
         }
         switch section {
         case .map:
-            filterSelectionDelegate?.filterContainerViewController(filterContainerViewController: self, navigateToMapFor: filterInfo)
+            navigator?.navigate(to: .map(filterInfo: filterInfo, parent: self))
             break
         case .all:
             toggleSelectAllSelection()
             updateAllVisibleCells()
+            let showButton = selectionDataSource.selectionState(filterInfo) != .none
+            showApplyButton(showButton)
         case .values:
-            if let listItem = listItems[safe: indexPath.row] {
-                if listItem.showDisclosureIndicator {
-                    didSelectDrillDownItem(listItem, at: indexPath)
+            guard let listItem = listItems[safe: indexPath.row] else {
+                return
+            }
+            if listItem.showDisclosureIndicator {
+                didSelectDrillDownItem(listItem, at: indexPath)
+            } else {
+                toggleSelection(for: listItem)
+                if filterInfo.isMultiSelect {
+                    updateCellIfVisible(at: indexPath)
+                    updateSelectAllCellIfVisible()
                 } else {
-                    toggleSelection(for: listItem)
-                    if filterInfo.isMultiSelect {
-                        updateCellIfVisible(at: indexPath)
-                        updateSelectAllCellIfVisible()
-                    } else {
-                        updateAllVisibleCells()
-                    }
+                    updateAllVisibleCells()
                 }
+                let showButton = selectionDataSource.selectionState(filterInfo) != .none
+                showApplyButton(showButton)
```

Comment:
Again I would move showApplyButton(...) into toggle function and no need to check selectionDataSoure